### PR TITLE
Add PHP 7.2 support, drop HHVM

### PR DESCRIPTION
**Pull request by Slamdunk**
- [x] Adopt recent `.travis.yml` ZF standards ([zend-session/.travis.yml](https://github.com/zendframework/zend-session/blob/develop/.travis.yml) picked)
- [x] Drop HVVM
- [x] Test PHP 7.2
- [x] Add partial PHP 7.2 support
    1. `idn_to_ascii(): INTL_IDNA_VARIANT_2003 is deprecated`
    1. `count(): Parameter must be an array or an object that implements Countable`

### CODE REVIEW

**Comment by froschdesign** [comment]
> PHP Fatal error:  Declaration of Zend\Session\AbstractContainer::offsetGet($key) must be compatible with & Zend\Stdlib\ArrayObject::offsetGet($key) in /home/travis/build/zendframework/zend-validator/vendor/zendframework/zend-session/src/AbstractContainer.php on line 27

https://travis-ci.org/zendframework/zend-validator/jobs/246138892#L347

**Comment by Ocramius** [comment]
@froschdesign https://github.com/zendframework/zend-session/pull/79

**Comment by Slamdunk** [comment]
Already fixed with https://github.com/zendframework/zend-session/pull/78, in fact `DEPS=latest` build passes:
https://travis-ci.org/zendframework/zend-validator/jobs/246138893

**Comment by froschdesign** [comment]
@Ocramius and @Slamdunk 
Right! (I was a bit too fast :see_no_evil: )

File: src/File/Upload.php [hunk 1]
```diff
@@ -109,7 +110,7 @@ public function getFiles($file = null)
      */
     public function setFiles($files = [])
     {
-        if (count($files) === 0) {
+        if ((is_array($files) || $files instanceof Countable) && count($files) === 0) {
```
Comment (Slamdunk) [inline src/File/Upload.php]: #### Side note
This bug highlights a higher priority bug in this method: the argument type isn't checked.
But to lessen the complexity of the PR and to speed up the next release, here I'm doing the minimum to get the code to work **exactly** like before.

Comment (Slamdunk) [inline src/File/Upload.php]: Well, it seems I was wrong: https://3v4l.org/Efkoc
So I guess a `$_FILES` test case is missing; condition changed.

**Comment by weierophinney** [comment]
Thanks, @Slamdunk.

**Pull request merged**